OSCreceiver.cpp

```diff
@@ -78,36 +83,76 @@ void OSCreceiver::ProcessMessage(const osc::ReceivedMessage &m, const IpEndpoint
 				msg->addStringArg(arg->AsStringUnchecked());
 			}
 		}
-		std::cout << "OSCreceiver::ProcessMessage " << m.AddressPattern() << " / " << _msg_queue.size() << std::endl;
-		_msg_queue.push_back(msg);
+		
+		emit_signal( "osc_message_received", add_gd_message( msg ) );
+		
 		check_queue();
+		
 	} catch( osc::Exception& e ){
+		
 		// any parsing errors such as unexpected argument types, or
 		// missing arguments get thrown as exceptions.
 		std::cout << "error while parsing message: "
 		<< m.AddressPattern() << ": " << e.what() << "\n";
+		
+	}
+	
+}
+
+const Dictionary& OSCreceiver::add_gd_message( gdOscMessage* msg ) {
+	
+	Dictionary d;
+
+	d["valid"] = true;
```

Comment:
Is it really useful to include this flag in the message? I am all for checking validity but if the message is invalid it should simply not be sent (and inform the user). It could be a (private?) member. This way we avoid the burden of verifying if the message is valid from GDscript.

Reply:
The getNextMessage should return something... In case of no more messages in the queue, I would usually return a nullptr... As i don't think that gdscript accepts pointers, what would be the solution to return something? A blank Dictionnary could be tested with "Dictionnary.empty()" >> http://docs.godotengine.org/en/3.0/classes/class_dictionary.html?highlight=Dictionary#class-dictionary-empty

Reply:
yes, I think testing against an empty dictionary is more intuitive.

Reply:
ok